pkg/syncer/storagepool/listener.go

```diff
@@ -114,17 +130,28 @@ func initListener(ctx context.Context, scWatchCntlr *StorageClassWatch, spContro
 			})
 			if err != nil {
 				log.Infof("Property collector session needs to be reestablished")
-				err = spController.vc.Connect(ctx)
+				reconnectFunc := func() error {
+					err = spController.vc.Connect(ctx)
+					if err != nil {
+						log.Warnf("Not able to establish connection with VC. Retrying...")
+						return err
+					}
+					return nil
+				}
+				err = common.ExponentialBackoff(reconnectFunc, 100, 2, 60000, 10)
```

Comment:
I still believe we don't need this. The PropertyCollector returns an error when the VC session is invalidated due to a creds change. Returning the error from here and exiting the thread will allow `managePCListenerInstance` to re-establish the connection. It is not expensive to re-establish the PropertyCollector session with a logged in VC client.